### DISCUSSION TITLE
Add a way to specify where cargo will spawn based on type

### DISCUSSION THREAD

**Discussion by gerakis** [FakeFishGames/Barotrauma#7943] [Ideas and suggestions]
I would like it if cargo spawners could be made specific to what type of cargo should spawn there. This way I could get ammo boxes to spawn in the armory, medical boxes to spawn in the infirmary, and normal cargo to spawn near the fabricator.

The way human spawners work is already like this. You can specify which class spawns where. You can also create general spawners that spawn everyone else. Just make cargo spawners work the same way. 

I think the best way to manage this would be by crate type. Medical items already spawn in a special medical crate. Just let sub builders determine where each of those crates ends up after a store visit.

**Comment by MaaiG** [comment]
I don’t agree. This is too easy IMHO.
 The crew must be able to move the cargo from the delivery point. Design your submarine to take into account the delivery lines!
Its will be better if you give the order to the crew "Placing cargo"!

**Reply by gerakis** [reply]
If you dont want your sub to work that way, just dont use it. Dont drag the rest of us down with you.

**Comment by MaaiG** [comment]
> If you dont want your sub to work that way, just dont use it. Dont drag the rest of us down with you.

I made my point.
It’s strange that you think of yourself as the opinion of many.
In addition. Please note the title of the section. It is called "Discussion".
Maybe you should think about saying "Dont drag the rest of us down with you". :)
_P.S. Casual game - sucks! Hardcore - forever!_ 👁️

P.S.2 Logistics can be a big part of the game. If promote it, it’s will be very interesting.
Any ship can have a character in charge of supply and logistics. He has extended powers including a special orders etc.
So there’s a lot to think about.
If someone wants to grow flowers, there’s someone who wants to do the cargo loading.

**Reply by gerakis** [reply]
This does not invalidate my point. You can build your sub however you want. The option should still be in the game.

**Comment by MateusJK15** [comment]
Good idea. And as Gerakis said, "if you don't want it, don't use it".

**Comment by MaaiG** [comment]
> This does not invalidate my point.

Obviously.

> if you don't want it, don't use it".

Of  course.

So in that case try to propose a mechanism for the realization of your idea. Describe and group all types of cargo into categories. Which categories of cargo. Which items fall into which category.
Suggest a specific option. Developers have so much to do that it will be much easier for them to consider a ready-made proposal rather than an idea that needs to be worked out.
For example: If I bought a `Harpoon Gun`, a `fuel rod`, two `Oxygen Tank` and `morphine`.
It all fits in one crate!
Do I get four crates in different places? Isn’t that too wasteful for a salesman? Just handing out crates?
Are you suggesting that objects appear immediately in some closet? Some cabinets? Or what?

Most people never really think it through but as I said **the logistics can be a big part of the game. If promote it, it’s will be very interesting. Any ship can have a character in charge of supply and logistics. He has extended powers including a special orders etc.**
So there’s a lot to think about. If someone wants to grow flowers, there’s someone who wants to do the cargo loading.
I don’t like the primitive approach to solving the problem.
That’s it. Good luck.

**Reply by WickedWolfy** [reply]
Should have read this post to the end before posting.
This one is on me.
Agree heavily with the last paragraph.

**Comment by WickedWolfy** [comment]
To add to the discussion.
A different option could be for the crew (read: bots) to be able to distribute items to their proper destinations.
This would also involve restocking locker boxes.

Example:
- Ordering 20 new bandages.
- Have crew/bots distribute these to small support lockers around the ship so at least one bandage would be available in all applicable boxes.

Issues with this would include tracking intelligence of supplies around the ship and possibly moving things around in a way that can't be tracked by the human players easily.
Also, how would it be determined what goes where and what storage needs what supplies.
AI will be a nightmare to program.

In this particular case I agree with @MaaiG that the current system should not be changed due to the sheer complexity of what needs to be built to support the OP's idea. With the existing simple tags the change is a little too great and too time-costly to implement.
Escape clause: @gerakis this game is, essentially, open-code. You can add this functionality on your own. Would be faster this way probably.

**Comment by Oukido** [comment]
I was thinking about the problem how to determine where items should be placed by bots quite a while now and came up with an idea that might be doable.
We already have autofill tags that control which items get placed inside containers by gamestart and also by the cleanup AI.
But during the gameplay this can drastically change since the need of items shifts depending on which technology we already have unlocked.
To react to this I suggest that every item that resides in a container grows a weight value for that slot over time. So for example if you put a bandage in a mineral container which by default the AI only places minerals, the bandage slot gets more and more weight until that one slot overweights the minerals and the AI starts to put bandages in that slot. (not all slots in that container). This way the AI would start filling up stacks of items that were placed in containers by the player over the course of the gameplay.

Furthermore I suggest to add a tag for crate shelves to only get filled with empty crates by the bots.
This way the items get sorted out of the crates and when this is done the crates themselves get cleaned up.

This is a bit of both worlds, regarding this discussion.

**Comment by Arctodites** [comment]
This would be such a useful QoL feature, especially for multiplayer games on public servers, where having to redo all the carrying and sorting after a restart (--> networking error / failed mission / griefer sabotage) can become tedious.  With a few additional tweaks to PvP level generation, it would also open up a world of possibilities for designing supply spawns in Outpost PvP maps.

It would also be great if we could designate in the sub editor whether or not crate shelves can be auto-filled for cargo transport missions, or whether they should be reserved for non-mission crate storage, and crate shelves flagged in the sub editor with a 'non-interactable' flag should not be counted towards the total crate storage count of a submarine in game menus.
